assets/js/base/components/checkout/form-step/index.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+const FormStep = ( {
+	id,
+	className,
+	stepNumber,
+	title,
+	description,
+	children,
+	secondaryAction,
+	isLastStep,
+} ) => {
+	return (
+		<div
+			className={ classnames( [
+				className,
+				'wc-components-checkout-step',
+				{ 'is-last-step': isLastStep },
+			] ) }
+			id={ id }
+		>
+			<StepNumber stepNumber={ stepNumber } />
+			<StepHeading title={ title } secondaryAction={ secondaryAction } />
+			<span className="wc-components-checkout-step__description">
+				{ description }
+			</span>
+			<div className="wc-components-checkout-step__content">
+				{ children }
+			</div>
+		</div>
+	);
+};
+
+const StepNumber = ( { stepNumber } ) => {
+	return (
+		<div className="wc-components-checkout-step__number">
+			{ stepNumber }
+		</div>
+	);
+};
+
+const StepHeading = ( { title, secondaryAction } ) => (
+	<div className="wc-components-checkout-step__heading">
+		<h4 className="wc-components-checkout-step__title">{ title }</h4>
+		{ secondaryAction && <SecondaryAction link={ secondaryAction } /> }
+	</div>
+);
+const SecondaryAction = ( { link } ) => (
+	<span className="wc-components-checkout-step__secondary-action">
+		{ link }
+	</span>
+);
```

Comment:
It's unclear what `SecondaryAction` is intended for.  The prop gives a clue that this would be a link, if so - then should this be an explicit button/a tag?

Reply:
This one was hard to implement for me, I needed to be flexible to accept text and a string and still be translatable so I did many iterations in it, it can’t be a direct button/a because it has both normal string and link in it.

SecondayAction is any behavior that might change the current view of the step or render the current view useless (use the same billing address in a shipping address step, login rather than creating an account..)

Reply:
I think there could be use cases where we (or extensions) might want to add a message here that is not a link, so I agree to change the `link` prop name, and maybe rename `SecondaryAction` too?

I'm really bad at naming things so I'm not sure I will be able to help here. But it kind of reminds me to the HTML [`<aside>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/aside) element, so maybe we could name it `<CheckoutStepAside>` or something like that? Regarding the prop, I think it could `children` or `content` considering it accepts any React node.

Reply:
> I'm really bad at naming things so I'm not sure I will be able to help here. But it kind of reminds me to the HTML <aside> element, so maybe we could name it <CheckoutStepAside> or something like that? Regarding the prop, I think it could children or content considering it accepts any React node.

👍 to @Aljullu 's comments.  How about we change prop name from `secondaryAction` to `StepHeadingContent` - all capitalized so it's clear it's a render prop.  I also agree with Albert that this can just be added as a child to `SecondaryAction` and in fact you can probably avoid the need for a `SecondaryAction` component at all and just embed the wrapping span directly in `StepHeading`.